src/test/regress/expected/multi_modifying_xacts.out

```diff
@@ -0,0 +1,429 @@
+ERROR:  distributed DDL commands cannot run inside a transaction block
+COMMIT;
+BEGIN;
+INSERT INTO labs VALUES (6, 'Bell Labs');
+\copy labs from stdin delimiter ','
+ERROR:  cannot open new connections after the first modification command within a transaction
+CONTEXT:  COPY labs, line 1: "7,E Corp"
+COMMIT;
+-- now, for some special failures...
+CREATE TABLE objects (
+	id bigint PRIMARY KEY,
+	name text NOT NULL
+);
+SELECT master_create_distributed_table('objects', 'id', 'hash');
+ master_create_distributed_table 
+---------------------------------
+ 
+(1 row)
+
+SELECT master_create_worker_shards('objects', 1, 2);
+ master_create_worker_shards 
+-----------------------------
+ 
+(1 row)
+
+-- test primary key violations
+BEGIN;
+INSERT INTO objects VALUES (1, 'apple');
+INSERT INTO objects VALUES (1, 'orange');
+ERROR:  duplicate key value violates unique constraint "objects_pkey_1200003"
+DETAIL:  Key (id)=(1) already exists.
+CONTEXT:  while executing command on localhost:57637
+COMMIT;
+-- data shouldn't have persisted...
+SELECT * FROM objects WHERE id = 1;
+ id | name 
+----+------
+(0 rows)
+
+-- and placements should still be healthy...
+SELECT count(*)
+FROM   pg_dist_shard_placement AS sp,
+	   pg_dist_shard           AS s
+WHERE  sp.shardid = s.shardid
+AND    sp.shardstate = 1
+AND    s.logicalrelid = 'objects'::regclass;
+ count 
+-------
+     2
+(1 row)
+
+-- create trigger on one worker to reject certain values
+\c - - - :worker_2_port
+CREATE FUNCTION reject_bad() RETURNS trigger AS $rb$
+    BEGIN
+        IF (NEW.name = 'BAD') THEN
+            RAISE 'illegal value';
+        END IF;
+
+        RETURN NEW;
+    END;
+$rb$ LANGUAGE plpgsql;
+CREATE CONSTRAINT TRIGGER reject_bad
+AFTER INSERT ON objects_1200003
+DEFERRABLE INITIALLY IMMEDIATE
+FOR EACH ROW EXECUTE PROCEDURE reject_bad();
+\c - - - :master_port
+-- test partial failure; worker_1 succeeds, 2 fails
+BEGIN;
+INSERT INTO objects VALUES (1, 'apple');
+INSERT INTO objects VALUES (2, 'BAD');
+WARNING:  illegal value
+CONTEXT:  while executing command on localhost:57638
+INSERT INTO labs VALUES (7, 'E Corp');
+COMMIT;
+-- data should be persisted
+SELECT * FROM objects WHERE id = 2;
+ id | name 
+----+------
+  2 | BAD
+(1 row)
+
+SELECT * FROM labs WHERE id = 7;
+ id |  name  
+----+--------
+  7 | E Corp
+(1 row)
+
+-- but one placement should be bad
+SELECT count(*)
+FROM   pg_dist_shard_placement AS sp,
+	   pg_dist_shard           AS s
+WHERE  sp.shardid = s.shardid
+AND    sp.nodename = 'localhost'
+AND    sp.nodeport = :worker_2_port
+AND    sp.shardstate = 3
+AND    s.logicalrelid = 'objects'::regclass;
+ count 
+-------
+     1
+(1 row)
+
+-- mark shards as healthy again; delete all data
+UPDATE pg_dist_shard_placement AS sp SET shardstate = 1
+FROM   pg_dist_shard AS s
+WHERE  sp.shardid = s.shardid
+AND    s.logicalrelid = 'objects'::regclass;
+DELETE FROM objects;
+WARNING:  modified 0 tuples, but expected to modify 2
```

Comment:
Hm. If I just move the `DELETE` to before the "mark all as healthy" operation, I can avoid this error. Doing that.